benchmark/run.py

```diff
@@ -0,0 +1,394 @@
+    atomic_smearing = half_cell / 5.0
+
+    # tight settings
+    cutoff_tight = half_cell
+    mesh_spacing_tight = atomic_smearing / 8.0
+    lr_wavelength_tight = atomic_smearing / 2.0
+
+    # light settings, roughly 4 times less computational costs
+    cutoff_light = cutoff_tight / 2.0
+    mesh_spacing_light = mesh_spacing_tight * 2.0
+    lr_wavelength_light = lr_wavelength_tight * 2.0
+
+    all_hypers[N] = {
+        "tight": {
+            "cutoff": cutoff_tight,
+            "mesh_spacing": mesh_spacing_tight,
+            "lr_wavelength": lr_wavelength_tight,
+            "atomic_smearing": atomic_smearing,
+        },
+        "light": {
+            "cutoff": cutoff_light,
+            "mesh_spacing": mesh_spacing_light,
+            "lr_wavelength": lr_wavelength_light,
+            "atomic_smearing": atomic_smearing,
+        },
+    }
+
+# -- system info --
+system = {
+    "platform": platform.platform(),
+    "cpu": platform.processor(),
+    "node": platform.node(),
+}
+
+if "cuda" in devices:
+    system["gpu"] = torch.cuda.get_device_name(0)
+
+
+# -- version info --
+cwd = os.path.dirname(os.path.abspath(__file__))
+try:
+    torch_pme_commit = subprocess.check_output(
+        ["git", "log", "--oneline", "-1"], cwd=cwd
+    ).decode("utf-8")
+    torch_pme_status = subprocess.check_output(
+        ["git", "status", "--porcelain"], cwd=cwd
+    ).decode("utf-8")
+except subprocess.CalledProcessError:
+    torch_pme_commit = "not found"
+    torch_pme_status = "not found"
+
+version = {
+    "torch": str(torch.__version__),
+    "torch-pme-commit": torch_pme_commit,
+    "torch-pme-status": torch_pme_status,
+    "torch-pme-version": torchpme.__version__,
+}
+
+
+# -- what do we have to do? --
+runs = {}
+
+for N, atoms in systems.items():
+    for h, hypers in all_hypers[N].items():
+        for d in devices:
+            if h == "tight" and d == "cpu":
+                # avoid very expensive work on CPU
+                continue
+
+            name = f"{N}_{h}_{d}"
+
+            runs[name] = {
+                "N": N,
+                "device": d,
+                "hypers": hypers,
+                "atoms": atoms.copy(),
+            }
+
+
+def compute_distances(positions, neighbor_indices, cell=None, neighbor_shifts=None):
+    """Compute pairwise distances."""
+    atom_is = neighbor_indices[:, 0]
+    atom_js = neighbor_indices[:, 1]
+
+    pos_is = positions[atom_is]
+    pos_js = positions[atom_js]
+
+    distance_vectors = pos_js - pos_is
+
+    if cell is not None and neighbor_shifts is not None:
+        shifts = neighbor_shifts.type(cell.dtype)
+        distance_vectors += shifts @ cell
+    elif cell is not None and neighbor_shifts is None:
+        raise ValueError("Provided `cell` but no `neighbor_shifts`.")
+    elif cell is None and neighbor_shifts is not None:
+        raise ValueError("Provided `neighbor_shifts` but no `cell`.")
+
+    return torch.linalg.norm(distance_vectors, dim=1)
```

Comment:
Is there a reason not to use external code or reusing the function we already defined in the testsuite? I am not a big fan of having duplicated code in the repo...

I am even okay moving the distance function as private function in the main repo, if you are afraid of abusing `sys.path.append`